EFCore.CheckConstraints/Internal/EnumCheckConstraintConvention.cs

```diff
@@ -105,38 +103,35 @@ private bool TryParseContiguousRange(ValueConverter? converter, IEnumerable valu
 
         var parameters = new object?[] { values, null, null };
 
-        var success = (bool)GetGenericTryGetMinMax(converter.ModelClrType).Invoke(null, parameters)!;
-
-        minValue = success ? parameters[1] : default;
-        maxValue = success ? parameters[2] : default;
-
-        return success;
-    }
-
-    private MethodInfo GetGenericTryGetMinMax(Type modelClrType)
-    {
-        var underlyingType = Enum.GetUnderlyingType(modelClrType);
+        var underlyingType = Enum.GetUnderlyingType(converter.ModelClrType);
 
         // ReSharper disable once InvertIf
-        if (!_cachedTryGetMinMaxMethodInfos.TryGetValue(underlyingType, out var methodInfo))
+        if (!_cachedTryGetMinMaxMethodInfos.TryGetValue(underlyingType, out var getMinMaxMethod))
         {
-            methodInfo = _tryGetMinMaxMethodInfo.MakeGenericMethod(underlyingType);
+            getMinMaxMethod = typeof(EnumCheckConstraintConvention).GetTypeInfo()
+                .GetDeclaredMethod(nameof(TryGetMinMax))!
+                .MakeGenericMethod(underlyingType);
 
-            _cachedTryGetMinMaxMethodInfos.Add(underlyingType, methodInfo);
+            _cachedTryGetMinMaxMethodInfos.Add(underlyingType, getMinMaxMethod);
         }
 
-        return methodInfo;
+        var success = (bool)getMinMaxMethod.Invoke(null, parameters)!;
+
+        minValue = success ? parameters[1] : default;
+        maxValue = success ? parameters[2] : default;
+
+        return success;
     }
 
-    private static bool TryGetMinMax<T>(IEnumerable values, out T minValue, out T maxValue)
+    private static bool TryGetMinMax<T>(object[] values, out T minValue, out T maxValue)
```

Comment:
Can this be `T[]`?

Reply:
I don't think so - the actual type of the array is `object[]` - it's created above with `Cast<object>().Distinct().ToArray()`. That cannot be just cast to e.g. `int[]`, which is why we must cast the individual members via `Cast<T>()` inside.

Reply:
It _would_ feel better if we were able to, but then we'd have to start working around the non-generic Array type and that's not very much fun.

Reply:
Yep.